frontend/src/static/js/components/webstatus-overview-filters.ts

```diff
@@ -192,6 +230,141 @@ export class WebstatusOverviewFilters extends LitElement {
+          columns.push(`${browserLabel} WPT ${channelLabel} Score`);
+        };
+
+        columnKeys.forEach(columnKey => {
+          const name = CELL_DEFS[columnKey].nameInDialog;
+          switch (columnKey) {
+            case ColumnKey.Name:
+              columns.push(name);
+              break;
+            case ColumnKey.BaselineStatus:
+              columns.push(name);
+              break;
+            case ColumnKey.StableChrome:
+            case ColumnKey.StableEdge:
+            case ColumnKey.StableFirefox:
+            case ColumnKey.StableSafari:
+            case ColumnKey.ExpChrome:
+            case ColumnKey.ExpEdge:
+            case ColumnKey.ExpFirefox:
+            case ColumnKey.ExpSafari:
+              pushBrowserChannelName(columnKey);
+              break;
+          }
+        });
+
+        // Convert array of feature rows into array of arrays of strings,
+        // in the same order as columns.
+        const rows = allFeatures.map(feature => {
+          const baselineStatus = feature.baseline?.status || '';
+          const browserImpl = feature.browser_implementations!;
+          const wptData = feature.wpt;
+          const row: string[] = [];
+
+          const pushBrowserChannelValue = (
+            browserColumnKey: BrowserChannelColumnKeys
+          ) => {
+            const {browser, channel} = getBrowserAndChannel(browserColumnKey);
+            const browserImplDate = browserImpl && browserImpl[browser]?.date;
+            const wptScore = wptData?.[channel]?.[browser]?.score;
+
+            if (channel === 'stable') {
+              row.push(browserImplDate || '');
+            }
+            row.push(String(wptScore) || '');
+          };
+
+          // Iterate over the current columns to get the values for each column.
+          for (const key of columnKeys) {
+            switch (key) {
+              case ColumnKey.Name:
+                row.push(feature.name);
+                break;
+              case ColumnKey.BaselineStatus:
+                row.push(baselineStatus);
+                break;
+              case ColumnKey.StableChrome:
+              case ColumnKey.StableEdge:
+              case ColumnKey.StableFirefox:
+              case ColumnKey.StableSafari:
+              case ColumnKey.ExpChrome:
+              case ColumnKey.ExpEdge:
+              case ColumnKey.ExpFirefox:
+              case ColumnKey.ExpSafari:
+                pushBrowserChannelValue(key);
+                break;
+            }
+          }
+          return row;
+        });
+
+        CSVUtils.downloadCSV(columns, rows, 'webstatus-feature-overview.csv')
+          .catch(error => {
+            toast(
+              `Save file error: ${error.message}`,
+              'danger',
+              'exclamation-triangle'
+            );
+          })
+          .finally(() => {
+            completedCallback && completedCallback();
+          });
+      })
+      .catch(error => {
+        toast(
```

Comment:
We could also move the `toast` call to the `renderExportButton` in a `catch`. Instead, we could throw an error here. 

This would allow us to separate the UI updates from logic (which would be helpful if we move the services architecture in the future)

Reply:
I lost the context of this suggestion, but I believe your change incorporates this.

Reply:
Done I believe